### PR TITLE
Issue #2951495 by mtodor: Add Split Text feature for paragraphs text fields

### PR DESCRIPTION
**Pull request by mtodor**

### CODE REVIEW

File: js/plugins/splittext/plugin.js [hunk 1]
```diff
@@ -0,0 +1,336 @@
+/**
+ * @file thunder-paragraph-features.split-text.js
```
Comment (chrfritsch) [inline js/plugins/splittext/plugin.js]: thunder-paragraphs-features.split-text.js

File: tests/src/FunctionalJavascript/ParagraphsFeaturesSplitTextTest.php [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+ * @group paragraphs_features
+ */
+class ParagraphsFeaturesSplitTextTest extends ParagraphsFeaturesJavascriptTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
```
Comment (chrfritsch) [inline tests/src/FunctionalJavascript/ParagraphsFeaturesSplitTextTest.php]: The modules, which are present in ParagraphsFeaturesSplitTextTest are not needed here

**Review by chrfritsch — CHANGES_REQUESTED** [review]

File: js/plugins/splittext/plugin.js [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+   * This could be provided in future as option where split text could work
+   * without any add mode, not just modal.
+   * -------------------------------------------------------------------------*
+   *
+   * @param {object} editor
+   *   CKEditor object.
+   */
+  Drupal.paragraphs_features.split_text.createNewParagraphOverDuplicate = function (editor) {
+    // Split text namespace.
+    var stNs = Drupal.paragraphs_features.split_text;
```
Comment (ol0lll) [inline js/plugins/splittext/plugin.js]: Maybe we could have this in a broader scope?

File: js/plugins/splittext/plugin.js [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+    // Set "cut" data ot new paragraph.
+    var $newRow = $originalRow.nextAll($originalRow.hasClass('odd') ? '.even' : '.odd');
+    var wrapperSelector = stNs.getEditorWrapperSelector(originalEditor);
+    var fieldIndex = $originalEditor.closest('div[data-drupal-selector="' + wrapperSelector + '"]').index();
+    var $newEditor = $($newRow.find('.paragraphs-subform > div')[fieldIndex]).find('textarea');
+    stNs.updateEditor($newEditor.attr('id'), stNs._tmp.newContent);
+
+    // Cleanup states.
+    stNs._tmp.split_trigger = false;
```
Comment (ol0lll) [inline js/plugins/splittext/plugin.js]: Please unify usage of tmpObject and stNs._tmp.

File: js/plugins/splittext/plugin.js [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+
+    // Delta field has to be cleaned up for proper working of add button. It
+    // will not make any impact on non modal add mode.
+    $originalRow.closest('table').siblings('.clearfix').find('input.paragraph-type-add-modal-delta').val('');
```
Comment (ol0lll) [inline js/plugins/splittext/plugin.js]: I guess we could just remove the clearfix selector.

**Review by ol0lll — CHANGES_REQUESTED** [review]

File: js/plugins/splittext/plugin.js (hunk 5 above)
Comment (mtodor) [reply js/plugins/splittext/plugin.js]: We need strict selector otherwise it could pickup nested paragraph add form.

Comment (mtodor) [reply js/plugins/splittext/plugin.js]: I didn't know that `find()` will work for all `siblings()`, but it does - so this is cleaned up too.

File: js/plugins/splittext/plugin.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+   * Get wrapper Drupal selector for CKEditor.
+   *
+   * @param {object} editor
+   *   CKEditor object.
+   *
+   * @return {string}
+   *   Returns CKEditor wrapper ID.
+   */
+  var getEditorWrapperSelector = function (editor) {
+    return editor.name.replace(/-[0-9]+-value(--[0-9A-Za-z_-]+)?$/, '-wrapper');
```
Comment (ol0lll) [inline js/plugins/splittext/plugin.js]: What about using something like $(editor.element).attr('data-drupal-selector').replace('-value', '-wrapper')

**Review by ol0lll — APPROVED** [review]

**Review by chrfritsch — APPROVED** [review]
Nice

**Pull request merged**